packages/model-viewer/src/features/annotation.ts

```diff
@@ -194,6 +205,41 @@ export const AnnotationMixin = <T extends Constructor<ModelViewerElementBase>>(
       hotspot.updateNormal(config.normal);
     }
 
+    /**
+     * This method returns the world position and normal of the point on the
+     * mesh corresponding to the input pixel coordinates given relative to the
+     * model-viewer element. The position and normal are returned as strings in
+     * the format suitable for putting in a hotspot's data-position and
+     * data-normal attributes. If the mesh is not hit, position returns the
+     * empty string.
+     */
+    getHitResult(pixelX: number, pixelY: number): HitResult {
+      const {width, height} = this[$scene];
+      this[$pixelPosition]
+          .set(pixelX / width, pixelY / height)
+          .multiplyScalar(2)
+          .subScalar(1);
+      this[$pixelPosition].y *= -1;
+      raycaster.setFromCamera(this[$pixelPosition], this[$scene].getCamera());
+      const hits = raycaster.intersectObject(this[$scene], true);
+
+      if (hits.length === 0) {
+        return {position: '', normal: ''};
+      }
+      const hit = hits[0];
+
+      const {x, y, z} = hit.point;
+      const hitPosition = `${x}m ${y}m ${z}m`;
```

Comment:
We typically report values from the imperative API as objects, not serialized string equivalents.

I recognize that it's convenient to get the serialized value for the purpose of assigning directly to an attribute (and that it is relatively cumbersome to construct a string manually), but it's possible for a user to construct a string from a structured object; it's a lot harder for them to parse the serialized string correctly if they want to access the values.

In the long run, a happy middleground might be to return specialized objects with `toString` implementations so that they can be easily serialized (and assigned to attributes directly w/o a manual serialization step).